core/templates/dev/head/pages/exploration_editor/translation_tab/AudioTranslationBarDirective.js

```diff
@@ -30,29 +30,31 @@ oppia.directive('audioTranslationBar', [
 
         $('.oppia-translation-tab').on('dragover', function(evt) {
           evt.preventDefault();
-          if (!scope.showDropArea) {
-            scope.showDropArea = true;
-            scope.$digest();
-          }
+          /* Don't show drag and drop area for user that is logged in
+            and cannot add translation. */
+          scope.dropAreaIsAccessible = !(!scope.userIsGuest &&
+            scope.translationIsDisallowed);
+          scope.$digest();
           return false;
         });
 
         $('.oppia-main-body').on('dragleave', function(evt) {
           evt.preventDefault();
           if (evt.pageX === 0 || evt.pageY === 0) {
-            scope.showDropArea = false;
+            scope.dropAreaIsAccessible = false;
             scope.$digest();
           }
           return false;
         });
 
         $('.oppia-translation-tab').on('drop', function(evt) {
           evt.preventDefault();
-          if (evt.target.classList.contains('oppia-drop-area-message')) {
+          if (evt.target.classList.contains('oppia-drop-area-message') &&
+            GLOBALS.can_translate && GLOBALS.userIsLoggedIn) {
```

Comment:
Should this be based on dropAreaIsAccessible too?

Reply:
Yes, I figured out a way to use that, while keeping all the functionality. Then next commit will have this.